racket-cas/racket-cas.rkt

```diff
@@ -848,174 +880,185 @@
+    (math-match v
+      [r          (- r)]
+      [(⊗ r u) (⊗ (- r) u)]
+      [_ (error)]))
+  (match u
+    [r #:when (and (real? r) (negative? r)) (values 0 (- r))]
+    [r #:when      (real? r)                (values r 0)]
+    [x #:when (symbol? x)                   (values x 0)] ; assume variables are positive
+    [(list* '+ us) (let loop ([us us] [pos '()] [neg '()])
+                     (match us
+                       ['()         (values (reverse-terms->sum pos)
+                                            (reverse-terms->sum neg))]
+                       [(cons u us) (if (negative-term? u)
+                                        (loop us         pos  (cons (negate-term u) neg))
+                                        (loop us (cons u pos)                       neg))]))]))
+                        
+
+  
+(module+ test
+  (displayln "TEST - partition-terms-after-sign")
+  (let-values ([(p n) (partition-terms-after-sign (normalize '(+ (exp 5) (* -2 x) z 2/3)))])
+    (check-equal? p (normalize '(+ 2/3 z (exp 5))))
+    (check-equal? n '(* 2 x)))
+  (let-values ([(p n) (partition-terms-after-sign -7)])
+    (check-equal? p 0)
+    (check-equal? n 7))
+  (let-values ([(p n) (partition-terms-after-sign  7)])
+    (check-equal? p 7)
+    (check-equal? n 0))
+  (let-values ([(p n) (partition-terms-after-sign '@i)])
+    (check-equal? p '@i)
+    (check-equal? n 0)))
+
 
 (define (term-numerator/denominator s)
-  (when debugging? (displayln (list 'term-numerator/denominator s)))
-  (term-numerator/denominator-impl (memoized-fractionize-number s))
-  (let-values ([(n d) (term-numerator/denominator-impl (fractionize-number s))])
-    (values (memoized-normalize n) (memoized-normalize d))) ; normalize cancels the effect of fractionize-number.
-  )
+  (numerator/denominator s))
 
-; Partition a product/terms into numerator and denominator
-(define (term-numerator/denominator-impl s)
-  (when debugging? (displayln (list 'term-numerator/denominator-impl s)))
-  (define nd term-numerator/denominator-impl)
-  (math-match s
-              [(Expt u (⊖ vp vn)) (values (Expt u vp) (Expt u vn))]
-              [(⊗ u v) (let-values ([(nu du) (nd u)] [(nv dv) (nd v)])
-                         (values (⊗ nu nv) (⊗ du dv)))]
-              [_ (values s 1)]))
 
 (module+ test
-  (let-values ([(n d)
-               (term-numerator/denominator (normalize '(* (/ x y) z 2/3)))])
+  (displayln "TEST - term-numerator/denominator")
+  (let-values ([(n d) (term-numerator/denominator (normalize '(* (/ x y) z 2/3)))])
     (check-equal? n '(* 2 x z))
-    (check-equal? d '(* 3 y))
-    )
-  )
+    (check-equal? d '(* 3 y))))
 
 ; used for quotient.
 ; first call together, then split up the term-numerator/denominator parts.
 ; differences can be indicated by the following tests.
-(define (numerator/denominator s)
-  (term-numerator/denominator (together s))
-  )
+(define (mma-numerator/denominator s)
+  (term-numerator/denominator (together s)))
+
```

Comment:
quotient can use mma-numerator/denominator

Reply:
I'll add an QuotientTogether so we have both options.